testutil-server/src/main/java/io/spine/testing/server/EmittedMessageSubject.java

```diff
@@ -150,8 +145,46 @@ public final S withType(Class<? extends M> messageClass) {
                                return match;
                            })
                            .collect(toImmutableList());
-            return check("withType()").about(factory())
-                          .that(filtered);
+            return check("withType(%s)", messageClass.getSimpleName())
+                    .about(factory())
+                    .that(filtered);
+        }
+    }
+
+    private @Nullable Iterable<T> messages() {
+        return actual;
+    }
+
+    /**
+     * Obtains the list of messages under assertion.
+     *
+     * @return an immutable copy of the {@code actual} messages
+     */
+    public List<T> actual() {
```

Comment:
I am not sure, but we tend to use the returning values of such methods as plain `List`s and then eventually wrap them into `ImmutableList`s over and over again. 

So if we can declare the `ImmutableList` as a type of the returning value, I would do so.